switch/ogre3d/ogre-switch.patch

```diff
@@ -0,0 +1,690 @@
++#define OGRE_PLATFORM_SWITCH 9
+     
+ #define OGRE_COMPILER_MSVC 1
+ #define OGRE_COMPILER_GNUC 2
+@@ -152,6 +153,8 @@ namespace Ogre {
+ #   endif
+ #elif defined(__ANDROID__)
+ #   define OGRE_PLATFORM OGRE_PLATFORM_ANDROID
++#elif defined(__SWITCH__)
++#   define OGRE_PLATFORM OGRE_PLATFORM_SWITCH
+ #else
+ #   define OGRE_PLATFORM OGRE_PLATFORM_LINUX
+ #endif
+diff --git a/OgreMain/include/OgreString.h b/OgreMain/include/OgreString.h
+index a4ff3ba..b1196ce 100644
+--- a/OgreMain/include/OgreString.h
++++ b/OgreMain/include/OgreString.h
+@@ -59,6 +59,16 @@ THE SOFTWARE.
+ #   include <xlocale.h>
+ #endif
+ 
++#if OGRE_PLATFORM == OGRE_PLATFORM_SWITCH
++#   include <stdlib.h>
++#   include <xlocale.h>
++#   include <strings.h>
++
++#   define strtod_l(ptr, end, l) strtod(ptr, end)
++#   define strtoul_l(ptr, end, base, l) strtoul(ptr, end, base)
++#   define strtol_l(ptr, end, base, l) strtol(ptr, end, base)
++#endif
++
+ namespace Ogre {
+     /** \addtogroup Core
+      *  @{
+diff --git a/OgreMain/src/OgreDeflate.cpp b/OgreMain/src/OgreDeflate.cpp
+index 56af65a..cedbdaa 100644
+--- a/OgreMain/src/OgreDeflate.cpp
++++ b/OgreMain/src/OgreDeflate.cpp
+@@ -155,6 +155,8 @@ namespace Ogre
+                 }
+ #elif OGRE_PLATFORM == OGRE_PLATFORM_APPLE_IOS || OGRE_PLATFORM == OGRE_PLATFORM_APPLE
+                 mTempFileName = macTempFileName();
++#elif OGRE_PLATFORM == OGRE_PLATFORM_SWITCH
++                OGRE_EXCEPT(Exception::ERR_INTERNAL_ERROR, "Temporary file name generation failed.", "DeflateStream::init");
+ #else
+                 char tmpname[] = "/tmp/ogreXXXXXX";
+                 if (mkstemp(tmpname) == -1)
+diff --git a/OgreMain/src/OgreFileSystem.cpp b/OgreMain/src/OgreFileSystem.cpp
+index 53b2af5..08446f4 100644
+--- a/OgreMain/src/OgreFileSystem.cpp
++++ b/OgreMain/src/OgreFileSystem.cpp
+@@ -32,7 +32,8 @@ THE SOFTWARE.
+ #if OGRE_PLATFORM == OGRE_PLATFORM_LINUX || OGRE_PLATFORM == OGRE_PLATFORM_APPLE || \
+     OGRE_PLATFORM == OGRE_PLATFORM_APPLE_IOS || \
+     OGRE_PLATFORM == OGRE_PLATFORM_ANDROID || \
+-    OGRE_PLATFORM == OGRE_PLATFORM_EMSCRIPTEN
++    OGRE_PLATFORM == OGRE_PLATFORM_EMSCRIPTEN || \
++    OGRE_PLATFORM == OGRE_PLATFORM_SWITCH
+ #   include "OgreSearchOps.h"
+ #   include <sys/param.h>
+ #endif
+diff --git a/OgreMain/src/OgreRoot.cpp b/OgreMain/src/OgreRoot.cpp
+index d5be1f5..37a5b3e 100644
+--- a/OgreMain/src/OgreRoot.cpp
++++ b/OgreMain/src/OgreRoot.cpp
+@@ -79,6 +79,10 @@ THE SOFTWARE.
+ #include "Android/OgreAndroidLogListener.h"
+ #endif
+ 
++#if OGRE_PLATFORM == OGRE_PLATFORM_SWITCH
++#include <switch.h>
++#endif
++
+ namespace Ogre {
+     //-----------------------------------------------------------------------
+     template<> Root* Singleton<Root>::msSingleton = 0;
+@@ -862,7 +866,11 @@ namespace Ogre {
+         // or break out by calling queueEndRendering()
+         mQueuedEnd = false;
+ 
++#ifdef __SWITCH__
++        while( !mQueuedEnd && appletMainLoop() )
```

Comment:
This smells of hackjob. Surely there is another way that other platforms are using to handle message-pump code and detect window close events?